app/operators/clinrev_pdf_operator.py

```diff
@@ -0,0 +1,286 @@
+from reportlab.lib.units import inch, cm
+from reportlab.lib import utils
+
+from reportlab.lib import colors
+from reportlab.lib.colors import white, black
+from reportlab.lib.fonts import tt2ps
+from reportlab.rl_config import canvas_basefontname as _baseFontName
+_baseFontNameB = tt2ps(_baseFontName, 1, 0)
+_baseFontNameI = tt2ps(_baseFontName, 0, 1)
+_baseFontNameBI = tt2ps(_baseFontName, 1, 1)
+
+
+@md.input("input_files", DataPath, IOType.DISK)
+@md.output("pdf_file", DataPath, IOType.DISK)
+@md.output("study_selected_series", List[StudySelectedSeries], IOType.IN_MEMORY)
+@md.env(pip_packages=["pydicom >= 2.3.0", "highdicom >= 0.18.2"])
+class ClinicalReviewPDFGenerator(Operator):
+
+    """Generates a PDF with Sag/Cor/Ax views for each structure"""
+
+    def compute(self, op_input: InputContext, op_output: OutputContext, context: ExecutionContext):
+
+        logging.info(f"Begin {self.compute.__name__}")
+        # get list of masks
+        input_path = op_input.get("input_files").path
+        dcm_input_path = input_path / 'dcm_input'
+        nii_seg_output_path = dcm_input_path / 'nii_seg_output'
+        nii_filenames = [join(nii_seg_output_path, f) for f in listdir(nii_seg_output_path) if
+                         isfile(join(nii_seg_output_path, f)) and '.nii' in f]
+
+        ct_nifti = dcm_input_path / 'input-ct-dataset.nii.gz'
+
+        logging.info(f"Creating PDF  ...")
+
+        # Get example CT dicom, get images and generate report
+        example_ct_file = [os.path.join(dcm_input_path, f) for f in os.listdir(dcm_input_path) if '.dcm' in f][0]
+        dcm_meta = pydicom.dcmread(example_ct_file, stop_before_pixels=True)
+        pdf_output_path = input_path / 'pdf'
+        Path(pdf_output_path).mkdir(parents=True, exist_ok=True)
+
+        ax_img_path, sag_img_path, cor_img_path = self.create_images_for_contours(dcm_meta=dcm_meta,
+                                                                                  output_path=pdf_output_path,
+                                                                                  ct_nifti_filename=ct_nifti,
+                                                                                  masks=nii_filenames)
+
+        pdf_filename = self.generate_report_pdf(dcm_meta,
+                                                output_path=pdf_output_path,
+                                                ax_img_path=ax_img_path,
+                                                sag_img_path=sag_img_path,
+                                                cor_img_path=cor_img_path,
+                                                nii_filenames=nii_filenames)
+
+        logging.info(f"DICOM Encapsulated PDF written to {pdf_filename}")
+        logging.info(f"PDF creation complete ...")
+        logging.info(f"End {self.compute.__name__}")
+        op_output.set(value=DataPath(pdf_filename), label='pdf_file')
+
+        return
```

Comment:
I don't think the `compute()` functions within an Operator class are supposed to have `return` statements.

Reply:
Removed